music_shop/user/serializers.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from rest_framework import serializers
+
+User = get_user_model()
+
+
+class RegisterSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ("email", "username", "password")
+        extra_kwargs = {"password": {"write_only": True}}
+
+    def create(self, validated_data):
+        user = User(
+            username=validated_data["username"],
+            email=validated_data["email"],
+            password=validated_data["password"],
+        )
+        user.set_password(validated_data["password"])
```

Comment:
Why should we do `.set_password`?

Reply:
We use `.set_password` because `.set_password` sets the user’s password to the given raw string, taking care of the password hashing.

Reply:
In this case, should we set password in line 18 ?

Reply:
No, we shouldn't, I've missed this line